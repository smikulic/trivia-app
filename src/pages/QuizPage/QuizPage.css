.QuizPage-question-section {
  padding: 1rem 4rem;
  height: 20rem;
}

.QuizPage-button-answers-section {
  display: flex;
  justify-content: space-evenly;
  margin: 1.5rem 0;
  width: 19rem;
}
