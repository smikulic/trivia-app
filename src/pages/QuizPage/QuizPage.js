import React from 'react';
import { useParams } from "react-router-dom";
import ButtonAnswer from '../../components/ButtonAnswer';
import StatusText from '../../components/StatusText';
import './QuizPage.css';

function QuizPage({ questions, totalQuestions, handleOnAnswer }) {
  const { questionId } = useParams();
  const currentQuestion = parseInt(questionId, 10);
  const nextQuestion = currentQuestion + 1;
  const question = questions[currentQuestion - 1];

  return (
    <React.Fragment>
      <div className="QuizPage-question-section">
        <h1>{question.category || 'N/A'}</h1>
        <h3>{question.question}</h3>
      </div>
      <div className="QuizPage-button-answers-section">
        <ButtonAnswer
          answerValue="true"
          handleOnClick={() => handleOnAnswer('True' === question.correct_answer, nextQuestion)}
        />
        <ButtonAnswer
          answerValue="false"
          handleOnClick={() => handleOnAnswer('False' === question.correct_answer, nextQuestion)}
        />
      </div>
      <StatusText
        statusNotice="Question"
        firstValue={currentQuestion}
        secondValue={totalQuestions}
      />
    </React.Fragment>
  );
}

export default QuizPage;
