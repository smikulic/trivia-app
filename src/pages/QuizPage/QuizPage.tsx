import React from 'react';
import { useParams } from "react-router-dom";
import ButtonAnswer from '../../components/ButtonAnswer';
import StatusText from '../../components/StatusText';
import ProgressBar from '../../components/ProgressBar';
import { IQuestion } from '../../types';
import { decodeHtml } from '../../utils/formatter';
import './QuizPage.css';
import Pagination from '../../components/Pagination';

interface IQuizPageProps {
  questionId: string,
  questions: IQuestion[],
  totalQuestions: number,
  handleOnAnswer: (arg1: boolean, arg2: number) => void,
  handleJumpToQuestion: (arg1: number) => void,
  answered: boolean,
  answers: any, // this needs to be boolean[] but there is an issues with TS and initial never[] state.
}

function QuizPage({
  questions,
  totalQuestions,
  handleOnAnswer,
  handleJumpToQuestion,
  answered,
  answers,
}: IQuizPageProps) {
  const { questionId } = useParams();
  const currentQuestion = questionId ? parseInt(questionId, 10) : 0;
  const nextQuestion = currentQuestion + 1;
  const question = questions[currentQuestion - 1];

  return (
    <React.Fragment>
      <ProgressBar percentage={currentQuestion / totalQuestions * 100} />
      <div className="QuizPage-question-section">
        <h1>{question.category || 'N/A'}</h1>
        <h3>{decodeHtml(question.question)}</h3>
      </div>
      <div className="QuizPage-button-answers-section">
        <ButtonAnswer
          answerValue="true"
          handleOnClick={() => handleOnAnswer('True' === question.correct_answer, nextQuestion)}
          disabled={answered}
        />
        <ButtonAnswer
          answerValue="false"
          handleOnClick={() => handleOnAnswer('False' === question.correct_answer, nextQuestion)}
          disabled={answered}
        />
      </div>
      <Pagination
        totalPages={totalQuestions}
        totalClickable={answers ? answers.length + 1 : 0}
        handlePageClick={handleJumpToQuestion}
      />
      <StatusText
        statusNotice=""
        firstValue={currentQuestion}
        secondValue={totalQuestions}
        small={true}
      />
    </React.Fragment>
  );
}

export default QuizPage;
