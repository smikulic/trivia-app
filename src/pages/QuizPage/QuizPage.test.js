import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import QuizPage from './QuizPage';

const questionsMock = [
  { question: 'question1?', category: 'category1' },
  { question: 'question2?', category: 'category2' },
]

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
  useParams: () => ({
    questionId: '1',
  }),
}));

it('renders without crashing - all props', () => {
  const div = document.createElement('div');
  ReactDOM.render(<QuizPage questions={questionsMock} totalQuestions={10} handleOnAnswer={() => undefined} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer
    .create(<QuizPage questions={questionsMock} totalQuestions={10} handleOnAnswer={() => undefined} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
