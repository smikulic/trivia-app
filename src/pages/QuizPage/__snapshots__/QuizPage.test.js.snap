// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="progress-bar"
    style={
      Object {
        "width": "10%",
      }
    }
  />,
  <div
    className="QuizPage-question-section"
  >
    <h1>
      category1
    </h1>
    <h3>
      question1?
    </h3>
  </div>,
  <div
    className="QuizPage-button-answers-section"
  >
    <div
      className="button-answer true"
      onClick={[Function]}
    >
      true
    </div>
    <div
      className="button-answer false"
      onClick={[Function]}
    >
      false
    </div>
  </div>,
  <div
    className="pagination"
  >
    <svg
      className="disabled"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
      />
    </svg>
    <svg
      className="disabled"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
      />
    </svg>
    <svg
      className="disabled"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
      />
    </svg>
    <svg
      className="disabled"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
      />
    </svg>
    <svg
      className="disabled"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
      />
    </svg>
    <svg
      className="disabled"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
      />
    </svg>
    <svg
      className="disabled"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
      />
    </svg>
    <svg
      className="disabled"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
      />
    </svg>
    <svg
      className="disabled"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
      />
    </svg>
    <svg
      className="disabled"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
      />
    </svg>
  </div>,
  <h4>
    
     
    1
     / 
    10
  </h4>,
]
`;
